test/integration/verify/controls/azurerm_virtual_machine.rb

```diff
@@ -1,35 +1,39 @@
-resource_group        = attribute('resource_group',        default: nil)
-vm                    = attribute('windows_vm_name',       default: nil)
-vm_id                 = attribute('windows_vm_id',         default: nil)
-location              = attribute('windows_vm_location',   default: nil)
-tags                  = attribute('windows_vm_tags',       default: nil)
-os_disk_name          = attribute('windows_vm_os_disk',    default: nil)
-data_disk_names       = attribute('windows_vm_data_disks', default: nil)
-monitoring_agent_name = attribute('monitoring_agent_name', default: nil)
+resource_group            = attribute('resource_group',        default: nil)
+win_name                  = attribute('windows_vm_name',       default: nil)
+win_id                    = attribute('windows_vm_id',         default: nil)
+win_location              = attribute('windows_vm_location',   default: nil)
+win_tags                  = attribute('windows_vm_tags',       default: nil)
+win_os_disk_name          = attribute('windows_vm_os_disk',    default: nil)
+win_data_disk_names       = attribute('windows_vm_data_disks', default: nil)
+win_monitoring_agent_name = attribute('monitoring_agent_name', default: nil)
```

Comment:
It just dawned on me that we may have a dangerous pattern here. If we fail to import an attribute:
```
win_name                  = attribute('windows_vm_name',       default: nil)
```
e.g. `win_name` comes back as `nil`. Then the test:
```
 its('name')                       { should eq win_name }
```
`name` would return `nil` and assert that it's the same as `win_name`. Using this pattern would every test pass if you forgot to pass in the attribute file now?

Reply:
Well the test run bombs in various ways when I knock chunks out of the attributes file, or omit it from the InSpec invocation, but yeah it's all just accidental breakage, not failing tests.

I'm not sure this is a problem. If it is, what do you think a solution might be? Just adding more examples like `its('name') { should_not be_nil }` or `its('name') { should_not be_empty }`? Wouldn't it be needed for all of the attributes, then?

Reply:
You're right the test will still return the actual value if it's there. Even if we fed in that we expected a `nil` value we'd fail because the actual Azure calls would return a value. 

I'll resolve this conversation.